vsphere/resource_vsphere_host.go

```diff
@@ -157,55 +164,43 @@ func resourceVsphereHostRead(d *schema.ResourceData, meta interface{}) error {
 	log.Printf("Setting lockdown to %s", lockdownMode)
 	d.Set("lockdown", lockdownMode)
 
-	lm := license.NewManager(client.Client)
-	am, err := lm.AssignmentManager(context.TODO())
-	if err != nil {
-		return err
-	}
-	licenses, err := am.QueryAssigned(context.TODO(), hostID)
-	if err != nil {
-		return err
-	}
-
 	licenseKey := d.Get("license").(string)
-	licFound := false
-	for _, lic := range licenses {
-		if licenseKey == lic.AssignedLicense.LicenseKey {
-			licFound = true
-			break
+	if licenseKey != "" {
+		licFound, err := isLicenseAssigned(client.Client, hostID, licenseKey)
+		if err != nil {
+			return err
 		}
-	}
 
-	if !licFound {
-		d.Set("license", "")
+		if !licFound {
+			d.Set("license", "")
+		}
 	}
 
 	return nil
 }
 
 func resourceVsphereHostCreate(d *schema.ResourceData, meta interface{}) error {
+	err := validateFields(d)
```

Comment:
This way of implementing the validation works. You can also use `CustomizeDiff` which will run before the CRUD functions. Thats how we've done it in some other resources, but either is fine. Mainly just wanted to point out the possibility.

Reply:
Thanks for the advice!